lib/Server.js

```diff
@@ -804,6 +825,12 @@ Server.prototype.listen = function (port, hostname, fn) {
     if (fn) {
       fn.call(this.listeningApp, err);
     }
+
+    if (this.options.open) {
+      const suffix = (this.options.inline !== false || this.options.lazy === true ? '/' : '/webpack-dev-server/');
+      const uri = createDomain({ host: hostname, port }, this.listeningApp) + suffix;
+      openBrowser(uri, this.options, this.log);
```

Comment:
It is should be in listen event and remove logic for open from bin file

Reply:
> remove logic for open from bin file

I'm not sure what do you mean by this. The previous comment says that open logic should be kept in bin

Reply:
Also socksjs doesn't seem to register a 'listen' event. On what object should i listen for and where should i call it?

Reply:
@amilajack listen `express` events https://github.com/webpack/webpack-dev-server/blob/master/bin/webpack-dev-server.js#L408

Reply:
But if I listen with express what port should i listen on? Also i don't understand the reason for listening on another port with express. Can you please explain this.